rfd/0167-auto-updates-change-proposal.md

```diff
@@ -0,0 +1,198 @@
+4. Teleport Cloud must provide the necessary tools for a user to build their own automation to keep Teleport client software up to date. This should include a stable API that users can rely on to request the latest version of Teleport that is compatible with their Teleport control plane.
+
+### Deprecate the stable/cloud teleport-ent package
+Currently, Teleport does not have full control of Teleport agent versions. A user is able to install the teleport-ent package at any version, including versions that may be incompatible with their Teleport control plane. A user is also able to manually update a Teleport agent to an incompatible version of Teleport.
+
+Step 1: The Teleport updater must control the version of Teleport agents at installation. To accomplish this, all agent installations will be done through the Teleport updater. The Teleport updater will now support an install command. The install command accepts the necessary configuration and then installs the latest compatible version of Teleport.
+
+An example of what this new installation process might look like:
+```sh
+$ apt-get install teleport-ent-updater
+$ teleport-upgrade install --proxy=example.teleport.sh
+```
+
+Step 2: Teleport must provide an alternate method of securely downloading the latest compatible version of Teleport. The Teleport CDN already serves Teleport artifacts and their SHA256 checksums. To ensure authenticity of these Teleport downloads, Teleport will now need to sign these artifacts and provide the digital signatures. The public key required to verify the digital signature will be baked into the Teleport updater.
+
+Step 3: The Teleport updater must control all Teleport updates. To ensure this, the Teleport updater will no longer rely on the system package manager to install/update Teleport. Users will no longer be able to manually update Teleport through the system package manager. Instead, the Teleport updater will now download the Teleport artifacts from the Teleport CDN along with the SHA256 checksum and the digital signature. The Teleport updater will verify the Teleport artifact and install Teleport into the /var/lib/teleport directory.
+
+The Teleport updater must be able to resolve conflict for these two situations:
+1. If Teleport is already installed on the system via the system package manager, the Teleport updater must be able to migrate the Teleport service to use the new binary installed using the Teleport CDN.
+2. If Teleport is already installed into the /var/lib/teleport directory, a new installation of Teleport via the system package manager must not override the Teleport binary used by the Teleport service.
+
+Step 4: The Teleport documentation should be updated to include a new section with instructions about how a user can build their own update automation.
+
+### Agent Version Management
```

Comment:
Can you add the proposed API for this?

Seems like we need at least three fields:
- Version
- Upgrade time OR immediate (can be a specific timestamp -- Cloud can translate window -> time)
- Jitter duration

And out-of-scope for this RFD, but eventually:
- Client bucket ID, for staged rollouts (upgrader holds off until the ID matches locally configured value)